src/detect-stream_size.c

```diff
@@ -245,7 +246,9 @@ static DetectStreamSizeData *DetectStreamSizeParse (const char *streamstr)
     }
 
     /* set the value */
-    sd->ssize = (uint32_t)atoi(value);
+    if (ByteExtractStringUint32(&sd->ssize, 10, 0, (const char *)value) < 0) {
+        return 0;
```

Comment:
This should be treated as an error. Also, returning from here will leave `mode, arg` and `value` dangling (leak).